kivy/uix/widget.py

```diff
@@ -522,6 +525,22 @@ def collide_widget(self, wid):
             return False
         return True
 
+    def on_motion(self, etype, me):
+        if self.disabled or me.flags & uix.WIDGET_BEHAVIOR_DISABLED:
+            return False
+        if me.type_id not in self.motion_filter:
+            return False
+        filtered = self.motion_filter[me.type_id]
+        if filtered[0] is self and len(filtered) == 1:
+            return False
+        if me.flags & uix.FILTERED_DISPATCH:
+            widgets = filtered[1:] if filtered[0] is self else filtered[:]
```

Comment:
not sure i understand the intent of this code, the effect of the flags is not documented, while the first is pretty obvious, this one less so.

Reply:
Variable `filtered` can hold `self` if registered for motion event and/or any of `self` children which also registered. Therefore, if `filtered[0] is self and len(filtered) == 1` is `True` only `self` is registered and we can stop the dispatch of the event to the children and subclass will handle the event.

Flags are used to suggest dispatch behavior of the event and so `me.flags & uix.FILTERED_DISPATCH` means to dispatch to registered child widgets instead of to all children.